task/mgmt-gateway/src/mgs_gimlet.rs

```diff
@@ -674,6 +674,34 @@ impl HostFlashUpdate {
         }
     }
 
+    fn status(&self) -> Result<UpdateStatus, ResponseError> {
+        let status = self.buf.status();
+        match &status {
+            // `UpdateBuffer` never sets its status to "preparing", as it
+            // doesn't know anything about that stage.
+            UpdateStatus::Preparing { .. } => panic!(),
+            UpdateStatus::InProgress(sub_status) => {
+                if let Some(sector_erase) = self.sector_erase.as_ref() {
+                    // If we're still erasing sectors, we shouldn't have
+                    // ingested any chunks yet.
+                    assert!(sub_status.bytes_received == 0);
+
+                    sector_erase.progress().map(|progress| {
+                        UpdateStatus::Preparing(UpdatePreparationStatus {
+                            id: sub_status.id,
+                            progress: Some(progress),
+                        })
+                    })
```

Comment:
Weakly-held opinion: since this function returns a `Result`, early-exiting may be cleaner than `map`, e.g.
```suggestion
                    let progress = sector_erase.progress()?;
                    UpdateStatus::Preparing(UpdatePreparationStatus {
                            id: sub_status.id,
                            progress: Some(progress),
                    })
```

Reply:
Yes, that's much better, thanks.